sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/client/CachedSchemaRegistryClient.java

```diff
@@ -171,23 +194,35 @@ public SchemaRegistryObject getSchemaByGuid(String schemaId) {
     }
 
     @Override
-    public String getSchemaId(
-        String schemaGroup, String schemaName, String schemaString, String schemaType) {
+    public String getSchemaId(String schemaGroup, String schemaName, String schemaString, String schemaType) {
         if (schemaStringCache.containsKey(schemaString)) {
             logger.verbose("Cache hit schema string. Group: '{}', name: '{}'", schemaGroup, schemaName);
             return schemaStringCache.get(schemaString).getSchemaId();
         }
 
-        SchemaId schemaId;
+        GetIdBySchemaContentResponse response;
         try {
-            schemaId = this.restService.getIdBySchemaContent(schemaGroup, schemaName, schemaType, schemaString);
+            response = this.restService
+                .getIdBySchemaContentWithResponseAsync(schemaGroup, schemaName, schemaType, schemaString)
+                .block(HTTP_REQUEST_TIMEOUT);
```

Comment:
Same here, timeout should be an optional method param. You can still have a default in case it's not set.